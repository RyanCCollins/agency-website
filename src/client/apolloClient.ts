import { ApolloClient, createNetworkInterface } from 'apollo-client';

declare var window: {
  __INITIAL_STATE__: string,
};

const uri = process.env.API_URL || 'http://0.0.0.0:1338/api';

export default new ApolloClient({
  networkInterface: createNetworkInterface({
    uri,
  }),
  initialState: typeof window !== 'undefined' ? window.__INITIAL_STATE__ : null, // eslint-disable-line
  ssrForceFetchDelay: 100,
});
