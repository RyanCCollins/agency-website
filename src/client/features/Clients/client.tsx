import * as React from 'react';
import { Box, Image } from 'ui';

export default function Client({
  image,
  name,
}) {
  return (
    <Box style={{ flexBasis: '33%' }} pad="large" alignItems="center">
      <Image alt={`${name} logo`} src={image} imageSize="small" />
    </Box>
  );
}
