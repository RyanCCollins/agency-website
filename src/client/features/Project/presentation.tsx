import * as React from 'react';
import { Box, Headline, LoadingIndicator, Hero, Article } from 'ui';
import { Props as ParentProps } from './types';
import Hr from '../Portfolio/hr';

export default function ProjectPresentation({
  theme,
  project,
  loading,
  error,
}: ParentProps) {
  if (error) {
    return <p>{error}</p>;
  }
  if (loading || !project) {
    return (
      <Box
        alignItems="center"
        justifyContent="center"
        style={{ minHeight: 'calc(100vh - 100px)' }}
        backgroundColor={theme.offwhite}
      >
        <Box pad="large" alignItems="center">
          <LoadingIndicator isLoading />
        </Box>
      </Box>
    );
  }
  return (
    <Box alignItems="center" backgroundColor={theme.offwhite}>
      <Hero height={650} backgroundImage={project.image} />
      <Box pad="large" alignItems="center">
        <Headline color="#666" fontWeight={700}>{project.title}</Headline>
        <Hr color="#666" />
        <Article pad="large" content={project.content} />
      </Box>
    </Box>
  );
}
