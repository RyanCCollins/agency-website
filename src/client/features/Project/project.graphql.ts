import gql from 'graphql-tag';

export default gql`
  query Project($id: ID!) {
    project(id: $id) {
      id: _id
      title
      image
      content
    }
  }
`;
