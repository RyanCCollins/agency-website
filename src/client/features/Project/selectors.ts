import { State as GlobalState } from '../../state';
import { State, ErrorType } from './types';
import { createSelector, Selector } from 'reselect';

export const selectProject = () => (state: GlobalState): State => state.project;

export type SelectIsLoading = Selector<GlobalState, boolean>;
export const selectIsLoading: SelectIsLoading = createSelector(
  selectProject(),
  (project) => project.isLoading,
);

export type SelectError = Selector<GlobalState, ErrorType>;
export const selectError: SelectError = createSelector(
  selectProject(),
  (project) => project.error,
);

export type SelectData = Selector<GlobalState, string>;
export const selectData: SelectData = createSelector(
  selectProject(),
  (project) => project.data,
);
