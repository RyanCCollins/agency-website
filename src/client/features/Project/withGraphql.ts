import { graphql } from 'react-apollo';
import PROJECT_QUERY from './project.graphql';

export default graphql(PROJECT_QUERY, {
  skip: ({ params }) => !params || (params && !params.id),
  options: ({ params }) => ({
    variables: {
      id: params.id,
    },
  }),
  props: ({ data: { loading, project, error, refetch } }) => ({
    loading,
    project,
    error,
    refetch,
  }),
});
