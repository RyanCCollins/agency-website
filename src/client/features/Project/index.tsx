import * as React from 'react';
import { compose } from 'redux';
import { withTheme } from 'styled-components';
import Presentation from './presentation';
import withGraphql from './withGraphql';
import { PortfolioTypes } from '../../types';
import { ThemeColorMap } from '../../types';

export interface Props {
  project: PortfolioTypes.Project;
  loading: boolean;
  error: string;
  refetch: () => void;
  theme: ThemeColorMap;
}

class Project extends React.Component<Props, undefined> {
  public render() {
    return (
      <Presentation {...this.props} />
    );
  }
}

export default compose(withTheme, withGraphql)(Project);
