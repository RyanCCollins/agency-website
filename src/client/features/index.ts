/* GENERATOR-IMPORT */
import Project from './Project';
import Portfolio from './Portfolio';
import NavigationMenu from './NavigationMenu';
import Home from './Home';
import Layout from './Layout';

export {
  /* GENERATOR-EXPORT */
  Project,
  Portfolio,
  NavigationMenu,
  Home,
  Layout,
};
