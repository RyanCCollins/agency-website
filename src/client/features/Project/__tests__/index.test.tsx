import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import { ApolloProvider } from 'react-apollo';
import * as React from 'react';
import Project from '../index';
import theme from '../../../theming';
import client from '../../../apolloClient';
import store from '../../../store';
import ProjectPresentation from '../presentation';

describe('<Project />', () => {
  describe('when shallow rendered', () => {
    const wrapper = mount(
      <ApolloProvider store={store} client={client}>
        <ThemeProvider theme={theme}>
          <Project />
        </ThemeProvider>
      </ApolloProvider>,
    );

    it('should create a snapshot', () => {
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    it('should render a presentation component', () => {
      expect(wrapper.find(ProjectPresentation).length).toBe(1);
    });
  });
});
