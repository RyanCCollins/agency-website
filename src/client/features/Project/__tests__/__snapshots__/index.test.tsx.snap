// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> when shallow rendered should create a snapshot 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "addTypename": true,
      "dataId": undefined,
      "fieldWithArgs": [Function],
      "initialState": Object {},
      "middleware": [Function],
      "mutate": [Function],
      "networkInterface": HTTPFetchNetworkInterface {
        "_afterwares": Array [],
        "_middlewares": Array [],
        "_opts": Object {},
        "_uri": "http://0.0.0.0:1338/api",
      },
      "query": [Function],
      "queryDeduplication": false,
      "queryManager": QueryManager {
        "addTypename": true,
        "deduplicator": Deduplicator {
          "inFlightRequestPromises": Object {},
          "networkInterface": HTTPFetchNetworkInterface {
            "_afterwares": Array [],
            "_middlewares": Array [],
            "_opts": Object {},
            "_uri": "http://0.0.0.0:1338/api",
          },
        },
        "fetchQueryPromises": Object {},
        "idCounter": 1,
        "networkInterface": HTTPFetchNetworkInterface {
          "_afterwares": Array [],
          "_middlewares": Array [],
          "_opts": Object {},
          "_uri": "http://0.0.0.0:1338/api",
        },
        "observableQueries": Object {},
        "pollingTimers": Object {},
        "queryDeduplication": false,
        "queryDocuments": Object {},
        "queryIdsByName": Object {},
        "queryListeners": Object {},
        "reducerConfig": Object {
          "customResolvers": undefined,
          "dataIdFromObject": undefined,
        },
        "reduxRootSelector": [Function],
        "resultComparator": undefined,
        "resultTransformer": undefined,
        "scheduler": QueryScheduler {
          "inFlightQueries": Object {},
          "intervalQueries": Object {},
          "pollingTimers": Object {},
          "queryManager": [Circular],
          "registeredQueries": Object {},
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
        },
      },
      "reducerConfig": Object {
        "customResolvers": undefined,
        "dataIdFromObject": undefined,
      },
      "reduxRootKey": "apollo",
      "reduxRootSelector": null,
      "resetStore": [Function],
      "resultComparator": undefined,
      "resultTransformer": undefined,
      "setStore": [Function],
      "shouldForceFetch": false,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
      },
      "version": "local",
      "watchQuery": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "black1": "#0a0a0a",
        "black2": "#2d2d2d",
        "black3": "#555555",
        "dark1": "#13293D",
        "dark2": "#006494",
        "dark3": "#92374D",
        "error": "#ff324d",
        "gray1": "#696969",
        "gray2": "#7c7c7c",
        "gray3": "#828282",
        "light1": "#0072BB",
        "light2": "#ff9800",
        "light3": "#0C7489",
        "offwhite": "#f5f5f5",
        "ok": "#8cc800",
        "primary": "#03A9F4",
        "secondary": "#9c27b0",
        "warning": "#ffd602",
        "white1": "#fff",
        "white2": "#f3f3f3",
        "white3": "#e6e8ec",
      }
    }
  >
    <_class>
      <Apollo(Project)
        theme={
          Object {
            "black1": "#0a0a0a",
            "black2": "#2d2d2d",
            "black3": "#555555",
            "dark1": "#13293D",
            "dark2": "#006494",
            "dark3": "#92374D",
            "error": "#ff324d",
            "gray1": "#696969",
            "gray2": "#7c7c7c",
            "gray3": "#828282",
            "light1": "#0072BB",
            "light2": "#ff9800",
            "light3": "#0C7489",
            "offwhite": "#f5f5f5",
            "ok": "#8cc800",
            "primary": "#03A9F4",
            "secondary": "#9c27b0",
            "warning": "#ffd602",
            "white1": "#fff",
            "white2": "#f3f3f3",
            "white3": "#e6e8ec",
          }
        }
      >
        <Project
          theme={
            Object {
              "black1": "#0a0a0a",
              "black2": "#2d2d2d",
              "black3": "#555555",
              "dark1": "#13293D",
              "dark2": "#006494",
              "dark3": "#92374D",
              "error": "#ff324d",
              "gray1": "#696969",
              "gray2": "#7c7c7c",
              "gray3": "#828282",
              "light1": "#0072BB",
              "light2": "#ff9800",
              "light3": "#0C7489",
              "offwhite": "#f5f5f5",
              "ok": "#8cc800",
              "primary": "#03A9F4",
              "secondary": "#9c27b0",
              "warning": "#ffd602",
              "white1": "#fff",
              "white2": "#f3f3f3",
              "white3": "#e6e8ec",
            }
          }
        >
          <Presentation
            theme={
              Object {
                "black1": "#0a0a0a",
                "black2": "#2d2d2d",
                "black3": "#555555",
                "dark1": "#13293D",
                "dark2": "#006494",
                "dark3": "#92374D",
                "error": "#ff324d",
                "gray1": "#696969",
                "gray2": "#7c7c7c",
                "gray3": "#828282",
                "light1": "#0072BB",
                "light2": "#ff9800",
                "light3": "#0C7489",
                "offwhite": "#f5f5f5",
                "ok": "#8cc800",
                "primary": "#03A9F4",
                "secondary": "#9c27b0",
                "warning": "#ffd602",
                "white1": "#fff",
                "white2": "#f3f3f3",
                "white3": "#e6e8ec",
              }
            }
          >
            <Box
              alignItems="center"
              backgroundColor="#f5f5f5"
              justifyContent="center"
              style={
                Object {
                  "minHeight": "calc(100vh - 100px)",
                }
              }
            >
              <styled.div
                alignItems="center"
                backgroundColor="#f5f5f5"
                justifyContent="center"
                style={
                  Object {
                    "minHeight": "calc(100vh - 100px)",
                  }
                }
              >
                <div
                  className="jQTesV"
                  style={
                    Object {
                      "minHeight": "calc(100vh - 100px)",
                    }
                  }
                >
                  <Box
                    alignItems="center"
                    pad="large"
                  >
                    <styled.div
                      alignItems="center"
                      pad="large"
                    >
                      <div
                        className="fDxjjW"
                      >
                        <LoadingIndicator
                          isLoading={true}
                        >
                          <Box
                            alignItems="center"
                            boxSize={
                              Object {
                                "horizontal": "small",
                              }
                            }
                            justifyContent="center"
                            pad="medium"
                          >
                            <styled.div
                              alignItems="center"
                              boxSize={
                                Object {
                                  "horizontal": "small",
                                }
                              }
                              justifyContent="center"
                              pad="medium"
                            >
                              <div
                                className="ehCbyN"
                              >
                                <styled.svg
                                  viewBox="0 0 48 48"
                                >
                                  <svg
                                    className="fCfRLW"
                                    viewBox="0 0 48 48"
                                  >
                                    <styled.circle
                                      cx="24"
                                      cy="24"
                                      fill="none"
                                      r="21"
                                      stroke="#007acc"
                                      strokeWidth="6"
                                    >
                                      <circle
                                        className="pAoyR"
                                        cx="24"
                                        cy="24"
                                        fill="none"
                                        r="21"
                                        stroke="#007acc"
                                        strokeWidth="6"
                                      />
                                    </styled.circle>
                                  </svg>
                                </styled.svg>
                              </div>
                            </styled.div>
                          </Box>
                        </LoadingIndicator>
                      </div>
                    </styled.div>
                  </Box>
                </div>
              </styled.div>
            </Box>
          </Presentation>
        </Project>
      </Apollo(Project)>
    </_class>
  </ThemeProvider>
</ApolloProvider>
`;
